### PR TITLE
docs: new extension `vike-react-styled-jsx`

### PR DESCRIPTION
**Pull request by phonzammi**
Hi Rom, 

Please take a look and review this PR

### CODE REVIEW

File: docs/pages/styled-jsx/+Page.mdx [hunk 1]
```diff
@@ -1,13 +1,20 @@
...
-Example:
+## Manual integration
+
+If you use <Link href="/vike-react">`vike-react`</Link>, you can manually integrate styled-jsx by using:
+- <Link href="/onBeforeRenderHtml" doNotInferSectionTitle />
+- <Link href="/Wrapper" doNotInferSectionTitle />
+- <Link href="/onAfterRenderHtml" doNotInferSectionTitle />
+
+Example of using Vike with styled-jsx:
  - <Example timestamp="2023.07" repo="jeremypress/vite-ssr-styled-jsx" />
```
Comment (brillout) [inline docs/pages/styled-jsx/+Page.mdx:14]: Do the example use `vike-react` or only Vike core?

File: docs/pages/styled-jsx/+Page.mdx [hunk 2]
```diff
@@ -1,13 +1,20 @@
...
    > vite-plugin-ssr was the [previous name of Vike](https://vite-plugin-ssr.com/vike).
 
-To use Vike with `styled-jsx`:
+To use Vike with `styled-jsx` without <Link href="/vike-react">`vike-react`</Link>:
```
Comment (brillout) [inline docs/pages/styled-jsx/+Page.mdx:17]: :100: Good catch. Intervention-=2.

Let's create a new section for it ``## Without `vike-{react,vue,solid}``` (it's a common convention, see for example https://vike.dev/ssr#without-vike-react-vue-solid).

File: docs/pages/styled-jsx/+Page.mdx [hunk 3]
```diff
@@ -1,13 +1,20 @@
 import { Example, UiFrameworkExtension, CommunityNote } from '../../components'
 import { Link } from '@brillout/docpress'
 
-<CommunityNote tool="styled-jsx" url="https://github.com/vercel/styled-jsx" hasExtension={false} />
+<CommunityNote tool="styled-jsx" url="https://github.com/vercel/styled-jsx" hasExtension="react" />
```
Comment (brillout) [inline docs/pages/styled-jsx/+Page.mdx:4]: Let's add backticks. Its official name is with backticks (let's always look at how tool authors write their tool).

File: docs/pages/styled-jsx/+Page.mdx (hunk 1 above)
Comment (phonzammi) [reply docs/pages/styled-jsx/+Page.mdx:14]: That one is still using `vite-plugin-ssr`.

I forgot to add my example—I'll add it now

File: docs/pages/styled-jsx/+Page.mdx (hunk 3 above)
Comment (phonzammi) [reply docs/pages/styled-jsx/+Page.mdx:4]: Where should we add the backticks ?

Comment (brillout) [reply docs/pages/styled-jsx/+Page.mdx:4]: to styled-jsx

Comment (brillout) [reply docs/pages/styled-jsx/+Page.mdx:4]: Make the argument accept `JSX.Element`. `Intervention+=2`.

Comment (phonzammi) [reply docs/pages/styled-jsx/+Page.mdx:4]: Well, it's not from an argument but `pageContext.pageTitle`. I guess we can add an optional prop `toolTitle?: string | JSX.Element` for `<CommunityNote>`

Comment (brillout) [reply docs/pages/styled-jsx/+Page.mdx:4]: Ah, I see. How about we change it in `headingsDetached.ts` then and that's it.

Comment (phonzammi) [reply docs/pages/styled-jsx/+Page.mdx:4]: It doesn't work; the backticks are rendered literally. That's why I removed them in the first place

Screenshot 2024-12-17 at 22 41 17

Comment (brillout) [reply docs/pages/styled-jsx/+Page.mdx:4]: :+1: Alright, let's stick to styled-jsx without backticks then.

**Comment by phonzammi** [comment]
LGTM ?

File: docs/pages/styled-jsx/+Page.mdx [hunk 4]
```diff
@@ -1,13 +1,23 @@
...
+## Manual integration
+
+If you use <Link href="/vike-react">`vike-react`</Link>, you can manually integrate `styled-jsx` by using:
+- <Link href="/onBeforeRenderHtml" doNotInferSectionTitle />
+- <Link href="/Wrapper" doNotInferSectionTitle />
+- <Link href="/onAfterRenderHtml" doNotInferSectionTitle />
+
+Example of using Vike with styled-jsx:
+ - <Example timestamp="2024.12" repo="phonzammi/vike-react-styled-jsx-example" /> - Using <Link href="/vike-react">`vike-react`</Link>
  - <Example timestamp="2023.07" repo="jeremypress/vite-ssr-styled-jsx" />
```
[the remark this answers is not in the document]
Comment (phonzammi) [reply docs/pages/styled-jsx/+Page.mdx]: Okay https://github.com/vikejs/vike/pull/2028/commits/475d123c9a057a4f0cc5773f7b4c89c9948079cc

File: docs/pages/styled-jsx/+Page.mdx [hunk 5]
```diff
@@ -1,13 +1,21 @@
...
-   > vite-plugin-ssr was the [previous name of Vike](https://vite-plugin-ssr.com/vike).
+## Manual integration
+
+If you use <Link href="/vike-react">`vike-react`</Link>, you can manually integrate `styled-jsx` by using:
+- <Link href="/onBeforeRenderHtml" doNotInferSectionTitle />
+- <Link href="/Wrapper" doNotInferSectionTitle />
+- <Link href="/onAfterRenderHtml" doNotInferSectionTitle />
+
+Example of using Vike with `styled-jsx`:
+ - <Example timestamp="2024.12" repo="phonzammi/vike-react-styled-jsx-example" /> - Using <Link href="/vike-react">`vike-react`</Link>
```
Comment (brillout) [inline docs/pages/styled-jsx/+Page.mdx:14]:
```suggestion
 - <Example timestamp="2024.12" repo="phonzammi/vike-react-styled-jsx-example" />
```

File: docs/pages/styled-jsx/+Page.mdx [hunk 6]
```diff
@@ -1,13 +1,21 @@
...
- - <Example timestamp="2023.07" repo="jeremypress/vite-ssr-styled-jsx" />
-   > vite-plugin-ssr was the [previous name of Vike](https://vite-plugin-ssr.com/vike).
+## Manual integration
+
+If you use <Link href="/vike-react">`vike-react`</Link>, you can manually integrate `styled-jsx` by using:
+- <Link href="/onBeforeRenderHtml" doNotInferSectionTitle />
+- <Link href="/Wrapper" doNotInferSectionTitle />
+- <Link href="/onAfterRenderHtml" doNotInferSectionTitle />
+
+Example of using Vike with `styled-jsx`:
```
[the remark this answers is not in the document]
Comment (brillout) [reply docs/pages/styled-jsx/+Page.mdx:13]: Intervention++

**Comment by phonzammi** [comment]
LGTM ?

**Comment by brillout** [comment]
:+1: Let me have a look. Btw. the latest docpress now epxorts `parseMarkdownMini()` so we can now use this to parse `pageContext.pageTitle`.

File: docs/pages/styled-jsx/+Page.mdx [hunk 7]
```diff
@@ -1,13 +1,21 @@
...
+- <Link href="/onBeforeRenderHtml" doNotInferSectionTitle />
+- <Link href="/Wrapper" doNotInferSectionTitle />
+- <Link href="/onAfterRenderHtml" doNotInferSectionTitle />
 
 Example:
- - <Example timestamp="2023.07" repo="jeremypress/vite-ssr-styled-jsx" />
-   > vite-plugin-ssr was the [previous name of Vike](https://vite-plugin-ssr.com/vike).
+ - <Example timestamp="2024.12" repo="phonzammi/vike-react-styled-jsx-example" />
+
+## Without `vike-{react,vue,solid}`
```
Comment (brillout) [inline docs/pages/styled-jsx/+Page.mdx:16]: I don't think `styled-jsx` works with other UI frameworks? Intervention++

Comment (phonzammi) [reply docs/pages/styled-jsx/+Page.mdx]: updated : https://github.com/vikejs/vike/pull/2028/commits/7bcb969adc4d188c5e053ab56a6ab66e92b2dba7

**Comment by phonzammi** [comment]
> Btw. the latest docpress now epxorts `parseMarkdownMini()` so we can now use this to parse `pageContext.pageTitle`.

I've added the backticks.

Please take a look and let me know wdyt

**Comment by brillout** [comment]
Perfect! Thank you @phonzammi

**Comment by brillout** [comment]
I really like our new idea `## What it does` and the new convention `pageContext.{config||''}{toolName}`.

How about we propagate it to the other Vike extensions you created?

**Comment by brillout** [comment]
I'm looking at https://vike.dev/styled-jsx and I really like it &mdash; it's getting really good. Idea: how about we always create `## Extension` (if there aren't any extensions we can put "Contribtion welcome..." there).

**Comment by phonzammi** [comment]
> How about we propagate it to the other Vike extensions you created?

Sure, we can do that.

> Idea: how about we always create `## Extension`

I think we can split the `<CommunityNote>` and create `<ExtensionNote>`

**Comment by brillout** [comment]
> I think we can split the `<CommunityNote>` and create `<ExtensionNote>`

As you wish :+1: although I feel like if we keep it to be only one argument then we'll make our lives easier to keep it as one component. Components should be split by logic and reusability, not by their visual appearance.

**Pull request merged**